### PR TITLE
fix: allow provider staff to send follow-ups in broadcasts

### LINKED ISSUE #669: [BUG] Sending messages and broadcasts fails with "Failed to send" error
## Describe the Bug

Pressing the send button on a direct message or a broadcast results in a
"Failed to send" / "Failed to send broadcast" flash error popup. The message
is not delivered.

## To Reproduce

1. Log in as a provider or parent
2. Open a conversation or navigate to a broadcast page (`/provider/programs/:id/broadcast`)
3. Type a message and press Send
4. Observe the "Failed to send" error flash instead of successful delivery

## Expected Behavior

Message or broadcast is sent successfully. For direct messages, the message
appears in the conversation. For broadcasts, a success flash with recipient
count is shown and the user is navigated to the conversation view.

## Environment

- Component: backend
- Area: Messaging bounded context

## Code References

- `lib/klass_hero_web/live/messaging_live_helper.ex:194-204`: `handle_send_message/2` — calls `Messaging.send_message/4` and shows flash on `{:error, reason}`
- `lib/klass_hero_web/live/provider/broadcast_live.ex:72-100`: `send_broadcast` event handler — catch-all `{:error, reason}` shows "Failed to send broadcast"
- `lib/klass_hero_web/live/staff/staff_broadcast_live.ex:120-146`: Staff broadcast send handler — same catch-all
- `lib/klass_hero/messaging/application/commands/send_message.ex:79-102`: `SendMessage.execute/4` — `with` pipeline; possible failures: `:not_participant`, `:broadcast_reply_not_allowed`, DB/S3 errors
- `lib/klass_hero/messaging/application/commands/broadcast_to_program.ex:62-95`: `BroadcastToProgram.execute/4` — possible failure: `:missing_provider_id` when `scope.provider` is nil

## Additional Context

The error path is a catch-all `{:error, reason}` in the LiveViews — the
actual reason is logged via `Logger.error/2` but not surfaced to the user.
Checking server logs when reproducing should reveal the specific reason.

Key suspects from recent refactors:
- `5ea551d7` split `ForManagingParticipants` / `ForQueryingParticipants` — verify
  all use cases use the correct port (read vs. write) for each operation
- `BroadcastToProgram` returns `{:error, :missing_provider_id}` if
  `scope.provider` is nil — possible for staff scopes if `provider_id` opt
  is missing

### PR DESCRIPTION
**Pull request by MaxPayne89**
## Summary

- Adds a new `ForResolvingProviderStaff` ACL port + `ProviderStaffResolver` adapter (delegates to `Provider.get_active_staff_member_by_user/1`).
- Extends `verify_broadcast_send_permission/3` in `SendMessage` with a third permission branch: any active staff member of the broadcast's provider may post follow-ups.
- Wires the new adapter under `:messaging, :for_resolving_provider_staff` in `config/config.exs`.

## Why

Staff members who could initiate a broadcast couldn't send follow-ups in the conversation they just created. The runtime check only consulted the per-program `program_staff_participants` projection, while the staff broadcast LiveView's mount-time check (`StaffBroadcastLive.staff_assigned?/2`) permitted any active staff of the provider. The two checks must agree — otherwise staff get redirected into a broadcast view with a "Type a message" input that immediately fails.

## Review Focus

- `lib/klass_hero/messaging/application/commands/send_message.ex` — new `cond` branch and `active_staff_for_provider?/2` helper. The order matters: provider-owner check is cheapest, then projection lookup, then ACL call.
- `lib/klass_hero/messaging/adapters/driven/provider/provider_staff_resolver.ex` — pattern-matches `provider_id` to ensure the staff record is for *this* provider.
- Whether a new dedicated port is the right granularity vs. extending `ForResolvingProgramStaff` (kept separate because that port's docstring scopes it to the projection).

## Test Plan

- [x] Failing test added to `send_message_test.exs` confirming the bug
- [x] Test passes after the fix (`mix test test/klass_hero/messaging/application/commands/send_message_test.exs`)
- [x] Full messaging suite passes (`457` tests)
- [x] Full project suite passes (`mix precommit` — 4085 passed)
- [x] End-to-end UI verification via Playwright: staff user sent broadcast, then sent a follow-up message in the broadcast conversation without error

Closes #669

### CODE REVIEW

File: lib/klass_hero/messaging/adapters/driven/provider/provider_staff_resolver.ex [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        target: "provider",
+        operation: "active_staff_for_provider?"
+      )
+
+      case KlassHero.Provider.get_active_staff_member_by_user(user_id) do
+        {:ok, %{provider_id: ^provider_id}} -> true
+        _ -> false
+      end
+    end
+  end
```
[the remark this answers is not in the document]
Comment (MaxPayne89) [reply lib/klass_hero/messaging/adapters/driven/provider/provider_staff_resolver.ex]: Good catch — this was a real correctness gap, not just a hypothetical. `get_active_staff_member_by_user/1` returns only the most recently inserted active row, so a user with active staff_member records at multiple providers would have been wrongly denied for non-latest providers' broadcasts.

Fixed in ae3821d0 by adding a dedicated `Provider.active_staff_for_provider?/2` predicate (port + query + repo + facade) that filters by both `provider_id` and `user_id` via `Repo.exists?/1`. The resolver now delegates to it directly.

Skipped the `function_exported?/2` shotgun pattern — we control the Provider facade, so a single explicit function is cleaner than runtime API discovery.

Added a regression test in send_message_test.exs covering the multi-provider edge case.
